src/mlpack/tests/gan_test.cpp

```diff
@@ -174,7 +175,7 @@ BOOST_AUTO_TEST_CASE(GANMNISTTest)
   Log::Info << trainData.n_rows << "--------" << trainData.n_cols << std::endl;
 
   // Create the Discriminator network
-  FFN<CrossEntropyError<> > discriminator;
+  FFN<SigmoidCrossEntropyError<> > discriminator;
```

Comment:
`SigmoidCrossEntropy` is nothing but the standard `CrossEntropy` extended to the case of negative output (multiple classes). We use `CrossEntropy` in almost all of our `GAN` implementations, because most of them make use of `sigmoid` layer at the end of the `Discriminator` and most papers implement GANs this way as well. So we don't even need to worry about negative output. However, in this case, either we can keep in line with O'Reilly's implementation and use `SigmoidCrossEntropy`, or we can try using a `Sigmoid` layer at the end of the `Discriminator` in place of `ReLU` (or introduce a further layer). Probably this is the reason why the other implementations didn't have an issue with convergence.